### PR TITLE
Replace Int64 by std::int64_t

### PR DESCRIPTION
**Pull request by pepone**

### CODE REVIEW

File: cpp/include/IceUtil/Config.h [hunk 1]
```diff
@@ -163,25 +163,6 @@ namespace IceUtil
         noncopyable(const noncopyable&);
         const noncopyable& operator=(const noncopyable&);
     };
-
-//
-// Int64 typedef and ICE_INT64 macro for Int64 literal values
-//
-// Note that on Windows, long is always 32-bit
-//
-#if defined(_WIN32) && defined(_MSC_VER)
```
Comment (pepone) [inline cpp/include/IceUtil/Config.h:172]: we can use `std::int64_t` everywhere we need an 64 bit integer.

File: cpp/include/IceUtil/InputUtil.h [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-
-#include <IceUtil/Config.h>
-#include <string>
-
-namespace IceUtilInternal
-{
-    //
-    // Portable strtoll/_strtoi64
-    //
-    ICE_API IceUtil::Int64 strToInt64(const char*, char**, int);
```
Comment (pepone) [inline cpp/include/IceUtil/InputUtil.h:16]: replaced by `std::stoll` https://en.cppreference.com/w/cpp/string/basic_string/stol

File: cpp/include/IceUtil/OutputUtil.h [hunk 3]
```diff
@@ -13,8 +13,6 @@
 
 namespace IceUtilInternal
 {
-    ICE_API std::string int64ToString(IceUtil::Int64);
```
Comment (pepone) [inline cpp/include/IceUtil/OutputUtil.h:16]: Use `std::to_string` if required https://en.cppreference.com/w/cpp/string/basic_string/to_string

File: cpp/src/slice2cpp/Gen.cpp [hunk 4]
```diff
@@ -1329,7 +1329,7 @@ Slice::Gen::ForwardDeclVisitor::visitEnum(const EnumPtr& p)
         //
         if (explicitValue)
         {
-            H << " = " << int64ToString((*en)->value());
+            H << " = " << std::to_string((*en)->value());
```
Comment (pepone) [inline cpp/src/slice2cpp/Gen.cpp:1332]: We can probably remove the `to_string` call here, `<<` operator should handle `int64_t`

File: cpp/src/slice2rb/RubyUtil.cpp [hunk 5]
```diff
@@ -1302,14 +1301,8 @@ Slice::Ruby::CodeVisitor::writeConstantValue(
...
                 case Slice::Builtin::KindFloat:
                 case Slice::Builtin::KindDouble:
-                {
-                    _out << value;
-                    break;
-                }
                 case Slice::Builtin::KindLong:
                 {
-                    IceUtil::Int64 l;
-                    IceUtilInternal::stringToInt64(value, l);
```
Comment (pepone) [inline cpp/src/slice2rb/RubyUtil.cpp:1312]: AFAIK there was no point in this `stringToInt64` call.

File: cpp/src/IceSSL/OpenSSLCertificateI.cpp [hunk 6]
```diff
@@ -189,7 +189,7 @@ namespace
 #endif
             tzone = mktime(&localTime) - mktime(&gmTime);
         }
-        return chrono::system_clock::time_point(chrono::seconds(mktime(&tm) - IceUtil::Int64(offset) * 60 + tzone));
+        return chrono::system_clock::time_point(chrono::seconds(mktime(&tm) - int64_t(offset) * 60 + tzone));
```
Comment (externl) [inline cpp/src/IceSSL/OpenSSLCertificateI.cpp:192]:
```suggestion
        return chrono::system_clock::time_point(chrono::seconds(mktime(&tm) - int64_t{offset} * 60 + tzone));
```

File: cpp/src/Slice/Grammar.cpp [hunk 7]
```diff
@@ -3470,13 +3469,16 @@ yyparse(void)
...
-                        if (IceUtilInternal::stringToInt64(constant->value(), v))
+                        try
                         {
+                            std::int64_t v = std::stoll(constant->value(), nullptr, 0);
                             tok = make_shared<IntegerTok>();
                             tok->v = v;
                             tok->literal = constant->value();
                         }
+                        catch (const std::exception&)
+                        {
```
Comment (externl) [inline cpp/src/Slice/Grammar.cpp:3480]: Are there any concerns now that we eat the exception?

File: cpp/src/slice2php/Main.cpp [hunk 8]
```diff
@@ -1163,10 +1162,9 @@ CodeVisitor::writeConstantValue(const TypePtr& type, const SyntaxTreeBasePtr& va
                 }
                 case Slice::Builtin::KindLong:
                 {
-                    IceUtil::Int64 l;
-                    IceUtilInternal::stringToInt64(value, l);
+                    int64_t l = std::stoll(value, nullptr, 0);
                     // The platform's 'long' type may not be 64 bits, so we store 64-bit values as a string.
-                    if (sizeof(IceUtil::Int64) > sizeof(long) && (l < LONG_MIN || l > LONG_MAX))
+                    if (sizeof(int64_t) > sizeof(long) && (l < INT32_MIN || l > INT32_MAX))
```
Comment (externl) [inline cpp/src/slice2php/Main.cpp:1167]: Can we just use std::numeric_limits<int62_t>  for these min/max?

File: cpp/src/Slice/Parser.h [hunk 9]
```diff
@@ -34,31 +35,12 @@ namespace Slice
...
-
-#    endif
-
-#endif
-
-    const IceUtil::Int64 Int16Max = 0x7fff;
-    const IceUtil::Int64 Int16Min = -Int16Max - 1;
-    const IceUtil::Int64 ByteMax = 0xff;
-    const IceUtil::Int64 ByteMin = 0x00;
+    const std::int64_t Int32Max = INT32_MAX;
```
Comment (externl) [inline cpp/src/Slice/Parser.h:38]: Can we just use std::numeric_limits<int62_t>  for these min/max?

File: php/src/Types.cpp [hunk 10]
```diff
@@ -1850,7 +1868,7 @@ IcePHP::SequenceInfo::unmarshalPrimitiveSequence(
                 // TODO: can we remove this now
```
Comment (externl) [inline php/src/Types.cpp:1868]: Can we remove this now?

**Review by externl — APPROVED** [review]

**Review by bernardnormier — APPROVED** [review]

File: cpp/src/Slice/Grammar.cpp (hunk 7 above)
Comment (pepone) [reply cpp/src/Slice/Grammar.cpp:3480]: I don't think so, `tok` remains uninitialized and we report the error below.

File: php/src/Types.cpp (hunk 10 above)
Comment (pepone) [reply php/src/Types.cpp:1868]: I think we should remove the TODO, one 32 bit platform still in use is ARMHF

Comment (pepone) [reply php/src/Types.cpp:1868]: Keep in mind this is about the PHP numeric type, not the C++ int64_t

File: cpp/src/Slice/Parser.h (hunk 9 above)
Comment (pepone) [reply cpp/src/Slice/Parser.h:38]: I will check and update in a follow-up PR

**Pull request merged**